lib/posting_list/src/posting_list.rs

```diff
@@ -0,0 +1,86 @@
+use std::marker::PhantomData;
+
+use common::types::PointOffsetType;
+
+use crate::CHUNK_SIZE;
+use crate::value_handler::ValueHandler;
+use crate::view::PostingListView;
+use crate::visitor::PostingVisitor;
+
+/// Generic compressed posting list.
+///
+/// - `PostingList<Sized<()>>` when there are no values (unit type `()`), there are just compressed ids + remainders
+/// - `PostingList<Sized<V>>` when there are `SizedValue` values, each id includes one value stored within the
+///   fixed-sized chunks
+/// - `PostingList<VarSized<V>>` when there are `VarSizedValue` values, each id includes one value in the chunk,
+///   which points to the actual value in the var_size_data
+pub struct PostingList<H: ValueHandler> {
+    pub(crate) id_data: Vec<u8>,
+    pub(crate) chunks: Vec<PostingChunk<H::Sized>>,
+    pub(crate) remainders: Vec<PostingElement<H::Sized>>,
+    pub(crate) var_size_data: Vec<u8>,
+    pub(crate) last_id: Option<PointOffsetType>,
+    pub(crate) _phantom: PhantomData<H>,
+}
+
+#[derive(Clone, Debug)]
+pub struct PostingElement<S> {
+    pub(crate) id: PointOffsetType,
+    pub(crate) value: S,
+}
+
+#[derive(Debug, Clone)]
+#[repr(C)]
+pub struct PostingChunk<S: Sized> {
+    /// Initial data point id. Used for decompression.
+    pub initial_id: PointOffsetType,
+
+    /// An offset within id_data
+    pub offset: u32,
+
+    /// Sized values for the chunk.
+    pub sized_values: [S; CHUNK_SIZE],
+}
```

Comment:
Suggestion to consider:

1. Separate `(initial_id, offset)` from `sized_values` and put it into a separate array. So the binary search over ids would not `sized_values` into the disk cache.

2. Further step: instead of regular sorted array, use https://curiouscoding.nl/posts/static-search-tree/ for faster binary search.